### PR TITLE
group member cli, resolvers, authz hooks

### PR DESCRIPTION
**Pull request by golanglemonade**
Adds ability to add users to groups with a role of `ADMIN` or `MEMBER`: 

```
go run cmd/cli/main.go group get  -i 01HKZWM5EAV6GD0F7B28VF2W2D       
{
  "group": {
    "createdAt": "2024-01-12T15:28:03.157586-07:00",
    "createdBy": "01HKZV5QJ7NFENPXYQ131CFN2B",
    "description": "",
    "displayName": "Datum",
    "id": "01HKZWM5EAV6GD0F7B28VF2W2D",
    "logoURL": "",
    "members": [
      {
        "id": "01HKZWM5EP4T38FCMHBJXHJTEF",
        "role": "ADMIN",
        "user": {
          "firstName": "Sarah",
          "id": "01HKZV5QJ7NFENPXYQ131CFN2B",
          "lastName": "Funkhouser"
        }
      },
      {
        "id": "01HKZWT3CJGWWVJ9R43KX7HQVM",
        "role": "MEMBER",
        "user": {
          "firstName": "Matt",
          "id": "01HKZWR2WMBWV6VRGGMYXEHPTD",
          "lastName": "Anderson"
        }
      }
    ],

```

Tuple creation when adding group member: 
image

Resolves: https://github.com/datumforge/datum/issues/293

Follow-up: https://github.com/datumforge/datum/issues/413

### CODE REVIEW

File: internal/ent/enums/enums.go [hunk 1]
```diff
@@ -8,17 +8,17 @@ import (
 
 type Role string
 
-const (
+var (
 	RoleOwner  Role = "OWNER"
 	RoleAdmin  Role = "ADMIN"
 	RoleMember Role = "MEMBER"
 	Invalid    Role = "INVALID"
```
Comment (matoszz) [inline internal/ent/enums/enums.go:15]: why invalid?

File: schema/ent.graphql [hunk 2]
```diff
@@ -1656,9 +1655,9 @@ type OrgMembershipEdge {
 }
 """OrgMembershipRole is enum for the field role"""
 enum OrgMembershipRole @goModel(model: "github.com/datumforge/datum/internal/ent/enums.Role") {
-  OWNER
   ADMIN
   MEMBER
+  OWNER
```
Comment (matoszz) [inline schema/ent.graphql:1660]: would you also not need INVALID here?

File: internal/ent/enums/enums.go (hunk 1 above)
Comment (golanglemonade) [reply internal/ent/enums/enums.go:15]: use to fall through when converting string to enum: https://github.com/datumforge/datum/pull/421/files/c465ff4882329c93c7abc78d87cce4c959ace8fd#diff-205a636cbc77012f8203237623857512bd8aa8c970edbdc14d5f40a754ef7141R34-R45

File: schema/ent.graphql (hunk 2 above)
Comment (golanglemonade) [reply schema/ent.graphql:1660]: no, these are the roles that are valid for the org

**Review by matoszz — DISMISSED** [review]

**Review by matoszz — APPROVED** [review]

**Pull request merged**